mol_dqn/chemgraph/multi_obj_opt.py

```diff
@@ -0,0 +1,161 @@
+# coding=utf-8
+# Copyright 2018 The Google Research Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Optimizes QED of a molecule with DQN.
+
+This experiment tries to find the molecule with the highest QED
+starting from a given molecule.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import functools
+import json
+import os
+
+from absl import app
+from absl import flags
+
+
+from rdkit import Chem
+from rdkit import DataStructs
+from rdkit.Chem import AllChem
+from rdkit.Chem import Descriptors
+from rdkit.Chem import QED
+from dqn import deep_q_networks
+from dqn import molecules as molecules_mdp
+from dqn import run_dqn
+from dqn.tensorflow_core import core
+
+FLAGS = flags.FLAGS
+
+
+class MultiObjectiveRewardMolecule(molecules_mdp.Molecule):
+  """Defines the subclass of generating a molecule with a specific reward.
+
+  The reward is defined as a scalar
+    reward = weight * similarity_score + (1 - weight) *  qed_score
+  """
+
+  def __init__(self, target_molecule, similarity_weight, 
+               discount_factor, **kwargs):
+    """Initializes the class.
+
+    Args:
+      target_molecule: SMILES string. the target molecule against which we
```

Comment:
Please finish this docstring; you need to describe similarity_weight and discount_factor.

Reply:
Done